functional-test/src/test/groovy/org/rundeck/tests/functional/api/project/ConfigSpec.groovy

```diff
@@ -0,0 +1,497 @@
+
+        when: "#TEST: delete project"
+        def deleteResponse = doDelete("/project/${projectName}?deferred=false")
+        assert deleteResponse.successful
+
+        then:
+        deleteResponse.code() == 204
+    }
+
+    def "test-project-delete.sh"(){
+        given:
+        def client = getClient()
+        client.apiVersion = 14 // as the original test
+        def projectName = "testProjectDelete"
+        def projectDescription = "a description"
+        Object testProperties = [
+                "name": projectName,
+                "description": projectDescription,
+                "config": [
+                        "test.property": "test value",
+                ]
+        ]
+        def mapper = new ObjectMapper()
+
+        when: "TEST: POST /api/14/projects"
+        def response = client.doPost(
+                "/projects",
+                testProperties
+        )
+        assert response.successful
+        ProjectCreateResponse parsedResponse = mapper.readValue(
+                response.body().string(),
+                ProjectCreateResponse.class
+        )
+
+        then:
+        parsedResponse.name != null
+        parsedResponse.name == projectName
+
+        when: "#TEST: delete project"
+        def deleteResponse = doDelete("/project/${projectName}")
+        assert deleteResponse.successful
+
+        then:
+        deleteResponse.code() == 204
+    }
+
+    def "test-project-invalid"(){
+        given:
+        def client = getClient()
+        def projectName = "RhetoricalMiscalculationElephant"
+        def mapper = new ObjectMapper()
+
+        when:
+        def response = client.doGet("/project/$projectName")
+        def parsedBody = mapper.readValue(response.body().string(), Object.class)
+
+        then:
+        response.code() == 404
+        parsedBody.errorCode == "api.error.project.missing"
+        parsedBody.message == "Project does not exist: $projectName"
+
+    }
+
+    def "test-project-json"(){
+        given:
+        def client = getClient()
+        client.apiVersion = 14
+        def projectName = "RhetoricalMiscalculationElephant"
+        def mapper = new ObjectMapper()
+
+        when: "We check only the format of the response, regardless of the content"
+        def jsonResponseBody = client.doGet("/project/$projectName")
+        def responseString = jsonResponseBody.body().string()
+        def validJsonParse = mapper.readValue(responseString, Object.class)
+
+        then:
+        !isYamlValid(responseString)
+        validJsonParse != null
+
+    }
+
+    def "test-project-missing"(){
+        given:
+        def client = getClient()
+        def projectName = "RhetoricalMiscalculationElephant"
+
+        when:
+        def jsonResponseBody = client.doGet("/project/$projectName/resources")
+
+        then:
+        !jsonResponseBody.successful
+        jsonResponseBody.code() == 404
+
+    }
+
+    def "test-project-resources-404"(){
```

Comment:
this test doesn't match the validations from the original test. The endpoint to request should be  "${APIURL}/project/${proj}/resources"